### PR TITLE
fix(tokenrate): fixed json issue for bancor api response

### LINKED ISSUE #590: bug: fix json issue on bancor api on tokenrate package
(Failed to get USD: token: no available quote query service: 
- json: cannot unmarshal string into Go struct field bancorMarketData.data.rate of type float64 
- coingecko: 429429 Too Many Requests 
- coinmarketcap: 429429 Too Many Requests

### PR DESCRIPTION
**Pull request by cnlangzi**
### Changes
-

### Fixes
- fixed #590 

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
-

### CODE REVIEW

File: core/tokenrate/bancor.go [hunk 1]
```diff
@@ -69,27 +69,77 @@ func (qq *bancorQuoteQuery) getUSD(ctx context.Context, symbol string) (float64,
 	rate, ok := result.Data.Rate24hAgo["usd"]
 
 	if ok {
-		if rate == 0 {
+
+		if rate.Value == 0 {
 			rate, ok = result.Data.Rate["usd"]
 			if ok {
```
Comment (peterlimg) [inline core/tokenrate/bancor.go]: Seems should be `if !ok`

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: core/tokenrate/bancor.go (hunk 1 above)
Comment (cnlangzi) [reply core/tokenrate/bancor.go]: No. It is ‘ok’.

Comment (cnlangzi) [reply core/tokenrate/bancor.go]: refactored code.it should be more clean now.

**Review by peterlimg — APPROVED** [review]
lgtm

**Pull request merged**